### PR TITLE
When comparing SVG tests against their blessed version, ignore the first line

### PR DESCRIPTION
**Pull request by estebank**
`anstyle_svg` has some weird non-determinism in the width parameter, which makes tests blessed in one environment to fail in another. This is the *only* non-determinism detected so far, so we modify the diff check to ignore the first line of the SVG. In order for a test to fail/be updated by `--bless`, a different part of the file needs to also have changed. If other sources of non-determinism are found, we should back out of the "`--color=always` means `.svg`" change.

r? @compiler-errors

### CODE REVIEW

**Comment by estebank** [comment]
Context https://github.com/rust-lang/rust/pull/122029#issuecomment-2009130070

File: src/tools/compiletest/src/runtest.rs [hunk 1]
```diff
@@ -4652,7 +4655,13 @@ impl<'test> TestCx<'test> {
     }
 
     fn compare_output(&self, kind: &str, actual: &str, expected: &str) -> usize {
-        if actual == expected {
+        let are_different = match (self.force_color_svg(), expected.find('\n'), actual.find('\n')) {
+            // We ignore the first line of SVG files because
+            // the width parameter is non-deterministic.
```
Comment (klensy) [inline src/tools/compiletest/src/runtest.rs:4660]: add FIXME here?

**Review by lqd — APPROVED** [review]
I can reproduce the failure and this does fix it 👍

**Comment by klensy** [comment]
> I can reproduce the failure and this does fix it 👍

what command?
`` python x.py --stage=1 --jobs=4 t tests/ui/codemap_tests --force-rerun` can't repro on `x86_64-pc-windows-msvc`

**Comment by estebank** [comment]
> ahh `ignore-windows`
> But on windows still can't catch 750\751 px difference, only color diff.

The distinction is made because on windows we produce different colors than on all other platforms (cyan instead of light blue), which would require everyone that works with ui tests to have access to both a linux/mac *and* windows, in perpetuity.

**Comment by klensy** [comment]
> > what command?
> 
> A simple `./x test tests/ui/codemap_tests --force-rerun` on my `x86_64-unknown-linux-gnu`.

Still all ok, strange.

**Comment by estebank** [comment]
At work we noticed that this fix didn't make it into 1.78, which makes the .svg tests fail non-deterministically. If we do a dot-release, we should include this PR.

**Comment by apiraino** [comment]
Stable backport accepted (if a dot release will be planned) as per compiler team [on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/238009-t-compiler.2Fmeetings/topic/.5Bweekly.5D.202024-05-16/near/439031380). A backport PR will be authored by the release team at the end of the current development cycle.

@rustbot label +stable-accepted

**Pull request merged**